tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+`$Bname$`                   | Board name of the selected board.
+`$Bpack$`                   | Path to the pack that defines the selected board (BSP).
+`$Dname$`                   | Device name of the selected device.
+`$Dpack$`                   | Path to the pack that defines the selected device (DFP).
+`$Pack$`                    | Path to the CMSIS Pack Root directory.
+`$Pack(vendor.name)$`       | Path to specific pack with latest version. Example: `$Pack(NXP.K32L3A60_DFP)$`.
+`$Output(project)$`         | Output file of a related project that is defined in the `*.csolution.yml` file.
+
+ToDo: define directory structure; should we use `$Output(project[.build-type][+target-type])$`
+
+**Examples:**
+
+```yml
+groups:
+  - group:  "Main File Group"
+    defines:
+      - $Dname$                           # Generate a #define 'device-name' for this file group
+```
+
+```yml
+  - execute: Generate Image
+    os: Windows                           # on Windows run from
+    run: $DPack$/bin/gen_image.exe        # DFP the get_image tool
+```
+
+## Overall Structure
+
+
+The table below explains the top-level elements in each of the different `*.yml` input files.
+
+Keyword          | Allowed for following files..                  | Description
+:----------------|:-----------------------------------------------|:------------------------------------
+`default:`       | `[defaults].csettings.yml`, `*.csolution.yml`  | Start of the default section with [General Properties](#general-properties)
+`solution:`      | `*.csolution.yml`                              | Start of the [Collection of related Projects](#solution-collection-of-related-projects) along with build order.
+`project:`       | `*.cproject.yml`                               | Start of a Project along with properties - tbd; used in `*.cproject.yml`.
+`layer:`         | `*.clayer.yml`                                 | Start of a software layer definition that contains pre-configured software components along with source files.
+
+
+YML structure of a `*.cdefaults.yml` file
+
+```yml
+defaults:              # Start of default settings
+  build-types:         # Default list of build-types (i.e. Release, Debug)
+  compiler:            # Default selection of the compiler
+```
+
+YML structure of a `*.csolution.yml` file
+
+```yml
+target-types:          # List of build-types
+build-types:           # List of build-types
+
+compiler:              # Default selection of the compiler
+
+solution:              # Start of a project list that are related
+  - project:           # Reference to a project file
+```
+
+YML structure of a `*.cproject.yml` file
+```yml
+project:
+  compiler: AC6
+
+  groups:              # List of source file groups along with source files
+  components:          # List of software components used
+  layers:              # List of software layers that belong to the project
+```
+
+YML structure of a `*.clayer.yml` file
```

Comment:
I think it has more than this:

https://github.com/MDK-Packs/CB_Lab4Layer/blob/master/layer/Board/B-L475E-IOT01A/Board.clayer